jbpm-flow/src/main/java/org/jbpm/process/instance/context/variable/VariableScopeInstance.java

```diff
@@ -162,5 +168,41 @@ public void setContextInstanceContainer(ContextInstanceContainer contextInstance
     		this.variableInstanceIdPrefix = ((CompositeContextNodeInstance) contextInstanceContainer).getUniqueId();
     	}
 	}
+    
+    public void enforceRequiredVariables() {
+        VariableScope variableScope = getVariableScope();
+        for (Variable variable : variableScope.getVariables()) {
+            String name = variable.getName();
+            if (variableScope.isRequired(name)) {  
+                // check case file if it is prefixed
+                if (name.startsWith(VariableScope.CASE_FILE_PREFIX)) {
+                    if (!findCaseData(variable.getName())) {
```

Comment:
`variable.getName() `is already retrieved at `name`

```suggestion
                    if (!findCaseData(name)) {
```

Reply:
fixed